### PR TITLE
Deploy with Kustomize

### PR DESCRIPTION
**Pull request by arnaud-tincelin**
use Kustomize for deployments

### CODE REVIEW

**Review by Tatsinnit — CHANGES_REQUESTED** [review]
💡 Thank you for taking a start on this, fyi: (Just a key observation I will do a depth review latter)

* We need to keep `aks-periscope.yaml` file for separate PR once all this is understood and merged because it is consumed by various external tools like `az-cli` and `vscode` consume this `.yaml`.

File: docs/appendix.md [hunk 1]
```diff
@@ -1,30 +0,0 @@
-# Appendix
```
Comment (Tatsinnit) [inline docs/appendix.md:1]: 💡Idea share: Let's do all the docs update in separate PR, there is history to various docs, and we can cater them all separately.

Comment (arnaud-tincelin) [reply docs/appendix.md]: I only left the new file docs/kustomize.md and reverted the others

File: deployment/aks-periscope.yaml [hunk 2]
```diff
@@ -83,7 +85,7 @@ spec:
         - secretRef:
             name: azureblob-secret
         volumeMounts:
-        - mountPath: /aks-periscope
+        - mountPath: /log/aks-periscope
```
Comment (Tatsinnit) [inline deployment/aks-periscope.yaml:88]: 💡 minor: do we need to document this anywhere?

File: deployment/deamon-set.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
```
Comment (davidkydd) [inline deployment/deamon-set.yaml:1]: typo in file name - daemon vs deamon

File: deployment/aks-periscope.yaml (hunk 2 above)
Comment (davidkydd) [reply deployment/aks-periscope.yaml:88]: @arnaud-tincelin what is this change for? I recommend not changing this file at all due to production downstream dependencies

Comment (arnaud-tincelin) [reply deployment/aks-periscope.yaml:88]: What dependencies are your referring to?

Another PR about dockerfile (which must be done first) puts the binary at /aks-periscope so we can't mount a volume there anymore.

Anyway, the best option might be first to remove all dependencies to files in the code so that we don't need this volume at all.

Comment (davidkydd) [reply deployment/aks-periscope.yaml:88]: The az cli and vs code extensions both currently pull and deploy the live raw version of this file, and might make other assumptions about its structure:

https://github.com/Azure/vscode-aks-tools/blob/430e8931a06ca6fe59bc8bff9bb98954106237f8/src/commands/periscope/helpers/periscopehelper.ts#L118

https://github.com/Azure/azure-cli-extensions/blob/b613a52b445ee49509803c29622b4eb664d84501/src/aks-preview/azext_aks_preview/custom.py#L1991

I assume this design is to allow the versions of periscope which are deployed by the CLI / vs code to be updated via changes to the periscope repo only - however I don't particularly agree with it and consider it a break waiting to happen, as well as unnecessary friction and complexity  when trying to change periscope itself - would be happier to have dedicated files like /azcli/aks-periscope.yaml and /vscode/aks-periscope.yaml if they do reside in this repo.

Not sure if any of your planned changes would actually break the integrations, but I suggest treating this as a "golden file" and not touching it for the time being - perhaps the kustomize changes could reside in a deployment/kustomize or deployment/dev subdir for the time being?

Comment (arnaud-tincelin) [reply deployment/aks-periscope.yaml:88]: az cli & vscode are independent from aks-periscope so we are not going to add anything specific for any tool.
What we could do is to git tag the version of the aks-periscope.yaml file before this PR and the PR on dockerfile so that it matches with a docker image tag.
The only change for azcli & vscode would be in the URL where you would no longer target /latest/ (which is a good thing).
This way we can make any changes without fear to break what is currently in use. What do you think?

File: docs/kustomize.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- target:
+    group: apps
+    kind: DaemonSet
+    name: aks-periscope
+    version: v1
+  patch: |-
+    - op: add
+      path: '/spec/template/spec/containers/0/env/-'
+      value:
+        name: AZURE_BLOB_ACCOUNT_NAME
```
[the remark this answers is not in the document]
Comment (arnaud-tincelin) [reply docs/kustomize.md:18]: I have added an example file in /examples

**Review by Tatsinnit — APPROVED** [review]
👍 Cool, looks good. Thank you for reverting the `aks-periscope.yaml`  file as it is and as `iteration-1` these kustomize changes look good and not impacting any existing functionality.

We can update few things in iteration-2.
